components/raftstore/src/store/fsm/apply.rs

```diff
@@ -471,6 +471,21 @@ where
     /// If it returns true, all pending writes are persisted in engines.
     pub fn write_to_db(&mut self) -> bool {
         let need_sync = self.sync_log_hint;
+        // There may be put and delete requests after ingest request in the same fsm.
+        // To guarantee the correct order, we must ingest the pending_sst first, and
+        // then persist the kv write batch to engine.
+        if !self.pending_ssts.is_empty() {
```

Comment:
I am worried about this operation may break the correctness of the transaction. I do not think carefully yet. Could you please explain why it's correct or writing comments in code is better.

Reply:
When a fsm process an ingest command, it will ensure all the changes before this point are write to db by checking `self.last_sync_apply_index != self.apply_state.get_applied_index()` first. And then the sst meta will be added to `pending_ssts` for later ingesting. 
If some other write commands come before the above sst is ingested, they will be write to `kv_wb`. So in `write_to_db`, we must ingest the ssts first. Because the ingest command happens before the write commands in `kv_wb` for a specific fsm.

Reply:
The ingest command may be reordered with the normal write command.
For example, 1. ingest(contain key A), 2. write key A, 3. ingest.
The number 2 command will be written to kv db before ingest command due to batch.
So the key A will be wrong.
Is this case right? Or I miss the point?

Reply:
For a same fsm, there is at most one ingest command in `pending_ssts`. When processing no.3 command, we will flush the previous changes to db.